File: Sorting/python/binarySearch.py
def binarySearch(list, item):
	low = 0;		# 左区间
	high = len(list) - 1	# 右区间

	while low <= high:
		mid = low + (high - low) / 2
		guess = list[mid]
		if guess == item:		# 找到元素
			return mid
		elif guess > item:		# 比数字大
			high = mid - 1
		else:					# 比数字小
			low = mid + 1
	return None		# 没有找到指定元素

File: Sorting/c++/binarySearch.cpp
// 二分区间为左闭右闭的 [left, right], 传入的值为[0, n-1]
int binarySearch(int A[], int left, int right, int x)
{
	int mid;	// mid 为 left 和 right 的中点
	while(left < right){
		mid = left + (right - left) / 2;		// 使用left + (right - left) 来避免溢出
		if(A[mid] == x) return mid;		//找到x， 返回下标
		else if(A[mid] > x){		//中间的数大于 x
			right = mid - 1;
		}else{					//中间的数小于 x
			left = mid + 1;
		}
	}
	return left;
}